src/main/scala/com/scalapenos/riak/internal/RiakHttpClientHelper.scala

```diff
@@ -165,6 +165,25 @@ private[riak] class RiakHttpClientHelper(system: ActorSystem) extends RiakUriSup
   }
 
   // ==========================================================================
+  // Special operations
+  // ==========================================================================
+
+  def allKeys(server: RiakServerInfo, bucket: String): Future[RiakKeys] = {
+    httpRequest(Get(AllKeysUri(server, bucket))).map { response ⇒
+      response.status match {
+        case OK ⇒
+          response.entity.toOption.map { body ⇒
+            import spray.json._
+
+            body.asString.parseJson.convertTo[RiakKeys]
+          }.getOrElse(RiakKeys(Nil))
```

Comment:
Here you could use `response.entity.as[RiakKeys]` and drop the `spray.json` import, just like in `getBucketProperties` above.